crates/turborepo-lib/src/run/watch.rs

```diff
@@ -166,14 +169,23 @@ impl WatchClient {
                     run.abort();
                 }
 
+                let entrypoint_tasks = tasks
```

Comment:
I don't think this is right. IIUC, `tasks` is the set of tasks the user request via `turbo watch <task1> <task2>`. However, if there's a filter present, like `turbo watch test -F my-lib`, and we get a change for a package `my-dep`, we don't necessarily want to look for `my-dep#test`. We want any task in the task graph that is run in `my-dep`. In this case, it's reasonable that `my-dep#build` might be present, but `my-dep#test` might not be.